src/components/Tab/index.tsx

```diff
@@ -0,0 +1,23 @@
+import { FC } from 'react'
+import { NavLink } from 'react-router-dom'
+
+import { createIcon, getClassNames } from './core/tab.helpers'
+import { IProps } from './core/tab.typings'
+
+export const Tab: FC<IProps> = ({
+  text,
+  selected = false,
+  disabled = false,
+  iconStart,
+  link,
+  className,
+}) => {
+  const classNames = getClassNames({ selected, disabled, className })
+
+  return (
+    <NavLink to={link} className={classNames.join(' ')}>
+      {iconStart && createIcon({ icon: iconStart })}
+      {text}
+    </NavLink>
+  )
+}
```

Comment:
How does the reactivity around the selected/active state is handled?

As we use a router to navigate from configure to run and to result sub views, the tabs here are use to navigate on those routes.

By passing `selected` props to display the selected/active state, I don't know how we're gonna easily know which on should be active.

Could we use `isActive` prop from `NavLink` in combination with the route passed to `to={link}`.

If we need finer handle on the state, we could search for react-router-dom [api](https://github.com/remix-run/react-router/tree/main/docs). I suspect `useRouteMatch` to be a good candidate.

Sidenote: @alex-mngt I see that react-router-dom version is pinned to exactly `6`, is there a reason for it?

Reply:
I do agree with @moreirathomas, it seems that using IsActive prop is much mure appropriate and as I remember it, the granularity given by this prop should do the job.

There is also no reason to keep react router pinned to version 6, I don't know how it ended pinned to an exact version ^^